### ISSUE TITLE
`#[no_coverage]` renamed to `#[coverage(off)]` without a migration period

### ISSUE THREAD

**Issue by Zalathar** [taiki-e/cargo-llvm-cov#312]
**UPDATE(taiki-e)**: `#[no_coverage]` renamed to `#[coverage(off)]`. 
- When using `#[coverage(off)]` in the new nightly, replace `feature(no_coverage)` with `feature(coverage_attribute)`, `no_coverage` with `coverage(off)`, and coverage-helper 0.1 with coverage-helper 0.2.
- You can work around the "removed feature" error by pinning the nightly version.
- When dependencies use unstable features and the error occurs, you can work around them by passing `--no-cfg-coverage-nightly` or/and `--no-cfg-coverage` flags, in many cases.

---

See https://github.com/rust-lang/rust/pull/114656.

Apparently this has been intended for a while, but in August 2023 someone stepped up to actually implement the change, and it's now close to being merged into nightly.

When this happens, it will affect anyone who is following the advice at https://github.com/taiki-e/cargo-llvm-cov#exclude-function-from-coverage. Affected users will need to update their code to use the new attribute, or else temporarily pin to a nightly version from before the change.

**Comment by taiki-e** [comment]
It is unfortunate that this is being removed without a migration period.

As for [coverage-helper](https://github.com/taiki-e/coverage-helper), once https://github.com/rust-lang/rust/pull/114656 is merged, I plan to release a new 0.1.x version that does not generate `#[no_coverage]` in versions where it does not exist, and 0.2.0 that generate `#[coverage(off)]`.

I'm not sure what should be done about users who use `#[no_coverage]` without going through coverage-helper.
One way is to add a wrapper for `#[no_coverage]` to the coverage-helper. Combining the above plan with encouraging users to use this wrapper instead of `#[no_coverage]` would prevent users using the wrapper from running into this problem.

**Comment by taiki-e** [comment]
Anyway thanks for the heads up.

**Comment by Zalathar** [comment]
The PR is currently waiting for another review, since the earlier CI run failed. So it may not be too late to speak up and ask for a migration period.

It doesn’t seem like supporting both for a while would be particularly onerous.
